File: src/app/models/StandingInstruction.js

// Define collection and schema for StandingInstruction
export interface StandingInstruction {
    instructionId:
	type : string
    amount:
	type : Money
    nextExecutionDate:
	type : Date
    Account:
	type : Schema.Types.ObjectId
    Beneficiary:
	type : Schema.Types.ObjectId
    Frequency:
 	type : String
    Status:
 	type : String
#
    collection: 'standingInstructions'
}

File: bankingOnGolang/internal/model/StandingInstruction.go

package model

import (
    "time"
)

//==============================================================
// StandingInstruction Declaration
//==============================================================
type StandingInstruction struct {
    BaseModel
     InstructionId              string
    Amount          Money `gorm:"embedded;embeddedPrefix:standingInstruction_amount"`
    NextExecutionDate              time.Time
    AccountId           *uint
    Account             *Account `gorm:"foreignKey:AccountId"`
    BeneficiaryId           *uint
    Beneficiary             *ExternalAccount `gorm:"foreignKey:BeneficiaryId"`
    Frequency              StandingInstructionFrequency
    Status              StandingInstructionStatus

// parent associations as their child
    StandingInstructionsFromAccountId    *uint

}

